### PR TITLE
Make typing aggressive and mandatory, add missing/invalid annotations

### PR DESCRIPTION
**Pull request by ThePletch**
## Why

[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1683)

## This PR

* Mandates type annotations in all application code (but not in tests)
* Configures mypy to check types even if something is unannotated (e.g. in tests)
* Adds annotations where missing/inaccurate and fixes associated bugs

## Other notes

Unannotated third party libraries cannot have their types overridden, but you can manually declare a type when assigning the output of a third party library function to a variable, which allows workarounds for most issues presented by this.

### CODE REVIEW

File: orchestration/dagster_orchestration/hca_manage/main.py [hunk 1]
```diff
@@ -117,7 +119,7 @@ def check_data(args, host, parser):
         hca.check_for_all()
 
 
-def create_snapshot(args, host):
+def create_snapshot(args: argparse.Namespace, host: str) -> JobId:
```
Comment (ThePletch) [inline orchestration/dagster_orchestration/hca_manage/main.py:122]: the repetition of this argument signature makes me think that these methods should be wrapped in a class that accepts the args in its initializer, but that's out of scope for this PR. small bite?

File: orchestration/dagster_orchestration/hca_manage/manage.py [hunk 2]
```diff
@@ -382,7 +386,8 @@ def remove_all(self):
             dangling_project_refs=0
         )
 
-    def _process_rows(self, get_table_names, get_rids, soft_delete: bool, issue: str) -> int:
+    def _process_rows(self, get_table_names: Callable[[], set[str]], get_rids: Callable[[
+                      str], set[str]], soft_delete: bool, issue: str) -> int:
```
Comment (ThePletch) [inline orchestration/dagster_orchestration/hca_manage/manage.py:390]: autopep8 is responsible for this weird indentation. i'll fix it manually.

File: orchestration/dagster_orchestration/hca_orchestration/repositories.py [hunk 3]
```diff
@@ -1,9 +1,11 @@
-from dagster import repository
+from typing import Union
```
Comment (ThePletch) [inline orchestration/dagster_orchestration/hca_orchestration/repositories.py:1]: looking forward to the new python 3.10 syntax for union types (`Union[A, B]` -> `A | B`)

**Review by raaidbroad — APPROVED** [review]

File: orchestration/dagster_orchestration/hca_manage/main.py (hunk 1 above)
Comment (aherbst-broad) [reply orchestration/dagster_orchestration/hca_manage/main.py:122]: I think we need to refactor this script into separate scripts that are more targeted, it's become a bit unwieldy to use.

File: orchestration/dagster_orchestration/hca_manage/manage.py [hunk 4]
```diff
@@ -62,8 +66,12 @@ def __post_init__(self):
     def bigquery_client(self) -> bigquery.client.Client:
         return bigquery.Client(project=self.project)
 
+    @cached_property
+    def gcp_creds(self) -> google.auth.credentials.Credentials:
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_manage/manage.py:70]: ❓ What is using this? I can't find a reference to it

**Review by aherbst-broad — APPROVED** [review]

File: orchestration/dagster_orchestration/hca_manage/manage.py (hunk 4 above)
Comment (ThePletch) [reply orchestration/dagster_orchestration/hca_manage/manage.py:70]: Updated to make methods use it internally.

**Pull request merged**